central/complianceoperator/v2/benchmarks/datastore/singleton.go

```diff
@@ -0,0 +1,53 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/complianceoperator/v2/benchmarks/store/postgres"
+	"github.com/stackrox/rox/central/globaldb"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/defaults/complianceoperator"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+var (
+	once sync.Once
+
+	dataStore DataStore
+
+	complianceOperatorBenchmarkAdministrationCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.ComplianceOperator)))
+)
+
+func initialize() {
+	storage := postgres.New(globaldb.GetPostgres())
+	dataStore = New(storage)
+
+	setupBenchmarks(storage)
+}
+
+// Singleton provides the interface for non-service external interaction.
+func Singleton() DataStore {
+	if !features.ComplianceEnhancements.Enabled() {
+		return nil
+	}
+	once.Do(initialize)
+	return dataStore
+}
+
+func setupBenchmarks(s postgres.Store) {
+	benchmarks, err := complianceoperator.LoadComplianceOperatorBenchmarks()
+	utils.CrashOnError(err)
```

Comment:
**Important**

Please only log this error. The Benchmark data is not required to have a working ACS, crashing it in this scenario would lead to a customer escalation immediately because our product is unavailable.
Especially in our ACSCS this is a no-no.

For Policies IMHO it is another use-case whereas it is more the core-product. 
If Benchmark data is not provided, CO integration would still work.

Reply:
Make sense. I've changed to log the error 👍